lesson16/script.js

```diff
@@ -0,0 +1,188 @@
+    }
+    topicsArea.appendChild(fragmentTopics);
+}
+
+const renderImages = (newsDate) => {
+    const contentsArea  = document.getElementById('js-contents-area');
+    const imageArea     = document.createElement('div');
+    imageArea.className = "contents-item topics-img";
+
+    const fragmentTopicImg = document.createDocumentFragment();
+    for (const topics of newsDate){
+        const topicsImg = document.createElement('img');
+        topicsImg.id    = `${topics.categories + '-img' }`;
+        topicsImg.src   = topics.picture;
+        topicsImg.classList.add('topics-img__item');
+        fragmentTopicImg.appendChild(topicsImg);
+    }
+    contentsArea.appendChild(imageArea).appendChild(fragmentTopicImg);
+}
+
+const showNewIcon = () => {
+    const newIcon       = document.createElement('span');
+    newIcon.className   = "new-icon";
+    newIcon.textContent = "New";
+    return newIcon;
+}
+
+const showCommentIcon = () => {
+    const commentIcon     = document.createElement('img');
+    commentIcon.src       = "img/comment-dots-solid.svg";
+    commentIcon.className = "comment-icon";
+    return commentIcon;
+}
+
+const showCommentNumber = (commentLength) => {
+    const commentNumber       = document.createElement('span');
+    commentNumber.textContent = commentLength;
+    commentNumber.className   = "comment-number";
+    return commentNumber;
+}
+
+const renderArticles = (newsDate) => {
+    const contentsArea    = document.getElementById('js-contents-area');
+    const articleArea     = document.createElement('div');
+    articleArea.className = "contents-item articles";
+
+    const fragmentAritcle = document.createDocumentFragment(); 
+    for (const topics of newsDate){
+        const articleList = document.createElement('ul');
+        articleList.id    = `${topics.categories + "-articles"}`;
+        articleList.classList.add('article-list');
+
+        const articles = topics.articles;
+        for(const article of articles ){
+            const articleItem   = document.createElement('li');
+            const articleLink   = document.createElement('a'); 
+            articleItem.classList.add('article-list__item');
+            articleLink.textContent  = article.title;
+
+            const commentLength = article.comment.length; 
+            const articleDateMs = new Date(article.date).getTime();
+            const periodDayMs   = new Date().getTime() - (3*24*60*60*1000);
+
+
+            if(periodDayMs < articleDateMs){
+                articleLink.appendChild(showNewIcon());
+            }
+            
+            if(commentLength > 0){
+                articleLink.appendChild(showCommentIcon());
+                articleLink.appendChild(showCommentNumber(commentLength));
+            }
+
+            fragmentAritcle.appendChild(articleList).appendChild(articleItem).appendChild(articleLink);
+        };
+    }
+    contentsArea.appendChild(articleArea).appendChild(fragmentAritcle);    
+}
```

Comment:
IMO 
I thought the function renderArticles was deeply nested. It may be possible to separate them.
This is just my idea.

[SakyoCodesandbox](https://codesandbox.io/s/js-lesson16-forked-ghou2i?file=/script.)

Reply:
I think so too.
I will try the method of inheriting with split allocation!

Reply:
I separeted them.
https://github.com/nakagawamai/js-lesson/pull/18/commits/eb2847a64ddd5b2d5b75b54e717cf8d517cda2e4